crates/rover-config/src/layout.rs

```diff
@@ -0,0 +1,22 @@
+use anyhow::{Error, Result};
+use std::env;
+use std::path::PathBuf;
+
+pub fn rover_config() -> Result<PathBuf> {
+    let config_dir = match env::var("APOLLO_CONFIG").ok() {
```

Comment:
Nit: maybe `APOLLO_CONFIG_HOME` so it's clearly intended to be a directory?
https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html has all the XDG names for reference.